### PR TITLE
App post release fixes

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Upgrades app packages.
- Updates android Gradle version.
- Configures GitHub actions to check for changes in mobile folder.
- Removes unnecessary permissions from android manifest file.
- Updates the app to use packages from Google fonts.
- Applies latest imporements to the dart language

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### Related tickets
[MOB-89](https://airqoteam.atlassian.net/browse/MOB-89)

### CODE REVIEW

File: .github/workflows/dart-code-metrics.yaml [hunk 1]
```diff
@@ -4,7 +4,41 @@ on: [push]
...
+            if [[ $file == mobile/* ]]; then
+              echo "::set-output name=run_app_metrics::true"
+            fi
+
+          done < files.txt
+
+  metrics:
     name: App code metrics
+    needs: [check]
+    if: needs.check.outputs.run_app_metrics == 'true'
```
Comment (NoahNsimbe) [inline .github/workflows/dart-code-metrics.yaml:41]: cc @123MwanjeMike

Comment (123MwanjeMike) [reply .github/workflows/dart-code-metrics.yaml:41]: Thanks @NoahNsimbe

**Review by 123MwanjeMike — APPROVED** [review]

**Review by BenjaminSsempala — APPROVED** [review]
LGTM

**Review by Mnoble-19 — APPROVED** [review]
LGTM

**Pull request merged**